Model/OrderData.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class OrderData
+{
+    public function __construct(
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollection
+    ) {
+        $this->orderCollection  = $orderCollection;
+    }
+
+    public function getOrders($storeId)
+    {
+        $ordersArray   = [];
+        $orders        = $this->getOrderQuery($storeId);
+
+        foreach ($orders as $order) {
+            if(!trim($order->getCustomerEmail())) {
+                continue;
+            }
+            $orderItems = $order->getAllItems();
+            $orderProducts = [];
+            
+            foreach ($orderItems as $orderItem) {
+                $itemType = $orderItem->getProductType();
+                if ($itemType == 'configurable' || $itemType == 'bundle') { // exclude configurable/bundle parent product returned by getAllItems() method
+                    continue;
+                }
+                $orderProducts[] = [
+                    'productId' => $orderItem->getProductId(),
+                    'quantity'  => $orderItem->getQtyOrdered()
+                ];
+            }
+
+            $orderBillingData = $order->getBillingAddress();
+            $street           = $orderBillingData->getStreet();
+            $couponCode       = [];
```

Comment:
can't you just:
```php
$couponCode = $order->getCouponCode() || []; 
```

Reply:
This will be the correct construction:
`$couponCode       = $order->getCouponCode() ? [$order->getCouponCode()] : [];`